// @decky/ui/dist/globals/steam-client/system/AudioDevice.js
export var ESystemAudioDirection;
(function (ESystemAudioDirection) {
    ESystemAudioDirection[ESystemAudioDirection["Invalid"] = 0] = "Invalid";
    ESystemAudioDirection[ESystemAudioDirection["Input"] = 1] = "Input";
    ESystemAudioDirection[ESystemAudioDirection["Output"] = 2] = "Output";
})(ESystemAudioDirection || (ESystemAudioDirection = {}));
export var ESystemAudioPortDirection;
(function (ESystemAudioPortDirection) {
    ESystemAudioPortDirection[ESystemAudioPortDirection["Invalid"] = 0] = "Invalid";
    ESystemAudioPortDirection[ESystemAudioPortDirection["Input"] = 1] = "Input";
    ESystemAudioPortDirection[ESystemAudioPortDirection["Output"] = 2] = "Output";
})(ESystemAudioPortDirection || (ESystemAudioPortDirection = {}));
export var ESystemAudioPortType;
(function (ESystemAudioPortType) {
    ESystemAudioPortType[ESystemAudioPortType["Invalid"] = 0] = "Invalid";
    ESystemAudioPortType[ESystemAudioPortType["Unknown"] = 1] = "Unknown";
    ESystemAudioPortType[ESystemAudioPortType["Audio32f"] = 2] = "Audio32f";
    ESystemAudioPortType[ESystemAudioPortType["Midi8b"] = 3] = "Midi8b";
    ESystemAudioPortType[ESystemAudioPortType["Video32RGBA"] = 4] = "Video32RGBA";
})(ESystemAudioPortType || (ESystemAudioPortType = {}));
export var ESystemAudioChannel;
(function (ESystemAudioChannel) {
    ESystemAudioChannel[ESystemAudioChannel["Invalid"] = 0] = "Invalid";
    ESystemAudioChannel[ESystemAudioChannel["Aggregated"] = 1] = "Aggregated";
    ESystemAudioChannel[ESystemAudioChannel["FrontLeft"] = 2] = "FrontLeft";
    ESystemAudioChannel[ESystemAudioChannel["FrontRight"] = 3] = "FrontRight";
    ESystemAudioChannel[ESystemAudioChannel["LFE"] = 4] = "LFE";
    ESystemAudioChannel[ESystemAudioChannel["BackLeft"] = 5] = "BackLeft";
    ESystemAudioChannel[ESystemAudioChannel["BackRight"] = 6] = "BackRight";
    ESystemAudioChannel[ESystemAudioChannel["FrontCenter"] = 7] = "FrontCenter";
    ESystemAudioChannel[ESystemAudioChannel["Unknown"] = 8] = "Unknown";
    ESystemAudioChannel[ESystemAudioChannel["Mono"] = 9] = "Mono";
})(ESystemAudioChannel || (ESystemAudioChannel = {}));
