// @decky/ui/dist/globals/steam-client/system/Bluetooth.js
export var EBluetoothDeviceType;
(function (EBluetoothDeviceType) {
    EBluetoothDeviceType[EBluetoothDeviceType["Invalid"] = 0] = "Invalid";
    EBluetoothDeviceType[EBluetoothDeviceType["Unknown"] = 1] = "Unknown";
    EBluetoothDeviceType[EBluetoothDeviceType["Phone"] = 2] = "Phone";
    EBluetoothDeviceType[EBluetoothDeviceType["Computer"] = 3] = "Computer";
    EBluetoothDeviceType[EBluetoothDeviceType["Headset"] = 4] = "Headset";
    EBluetoothDeviceType[EBluetoothDeviceType["Headphones"] = 5] = "Headphones";
    EBluetoothDeviceType[EBluetoothDeviceType["Speakers"] = 6] = "Speakers";
    EBluetoothDeviceType[EBluetoothDeviceType["OtherAudio"] = 7] = "OtherAudio";
    EBluetoothDeviceType[EBluetoothDeviceType["Mouse"] = 8] = "Mouse";
    EBluetoothDeviceType[EBluetoothDeviceType["Joystick"] = 9] = "Joystick";
    EBluetoothDeviceType[EBluetoothDeviceType["Gamepad"] = 10] = "Gamepad";
    EBluetoothDeviceType[EBluetoothDeviceType["Keyboard"] = 11] = "Keyboard";
})(EBluetoothDeviceType || (EBluetoothDeviceType = {}));
