// @decky/ui/dist/globals/steam-client/system/Perf.js
export var EGPUPerformanceLevel;
(function (EGPUPerformanceLevel) {
    EGPUPerformanceLevel[EGPUPerformanceLevel["Invalid"] = 0] = "Invalid";
    EGPUPerformanceLevel[EGPUPerformanceLevel["Auto"] = 1] = "Auto";
    EGPUPerformanceLevel[EGPUPerformanceLevel["Manual"] = 2] = "Manual";
    EGPUPerformanceLevel[EGPUPerformanceLevel["Low"] = 3] = "Low";
    EGPUPerformanceLevel[EGPUPerformanceLevel["High"] = 4] = "High";
    EGPUPerformanceLevel[EGPUPerformanceLevel["Profiling"] = 5] = "Profiling";
})(EGPUPerformanceLevel || (EGPUPerformanceLevel = {}));
export var ESplitScalingFilter;
(function (ESplitScalingFilter) {
    ESplitScalingFilter[ESplitScalingFilter["Invalid"] = 0] = "Invalid";
    ESplitScalingFilter[ESplitScalingFilter["Linear"] = 1] = "Linear";
    ESplitScalingFilter[ESplitScalingFilter["Nearest"] = 2] = "Nearest";
    ESplitScalingFilter[ESplitScalingFilter["FSR"] = 3] = "FSR";
    ESplitScalingFilter[ESplitScalingFilter["NIS"] = 4] = "NIS";
})(ESplitScalingFilter || (ESplitScalingFilter = {}));
export var ESplitScalingScaler;
(function (ESplitScalingScaler) {
    ESplitScalingScaler[ESplitScalingScaler["Invalid"] = 0] = "Invalid";
    ESplitScalingScaler[ESplitScalingScaler["Auto"] = 1] = "Auto";
    ESplitScalingScaler[ESplitScalingScaler["Integer"] = 2] = "Integer";
    ESplitScalingScaler[ESplitScalingScaler["Fit"] = 3] = "Fit";
    ESplitScalingScaler[ESplitScalingScaler["Fill"] = 4] = "Fill";
    ESplitScalingScaler[ESplitScalingScaler["Stretch"] = 5] = "Stretch";
})(ESplitScalingScaler || (ESplitScalingScaler = {}));
export var ESystemServiceState;
(function (ESystemServiceState) {
    ESystemServiceState[ESystemServiceState["Unavailable"] = 0] = "Unavailable";
    ESystemServiceState[ESystemServiceState["Disabled"] = 1] = "Disabled";
    ESystemServiceState[ESystemServiceState["Enabled"] = 2] = "Enabled";
})(ESystemServiceState || (ESystemServiceState = {}));
export var EGraphicsPerfOverlayLevel;
(function (EGraphicsPerfOverlayLevel) {
    EGraphicsPerfOverlayLevel[EGraphicsPerfOverlayLevel["Hidden"] = 0] = "Hidden";
    EGraphicsPerfOverlayLevel[EGraphicsPerfOverlayLevel["Basic"] = 1] = "Basic";
    EGraphicsPerfOverlayLevel[EGraphicsPerfOverlayLevel["Medium"] = 2] = "Medium";
    EGraphicsPerfOverlayLevel[EGraphicsPerfOverlayLevel["Full"] = 3] = "Full";
    EGraphicsPerfOverlayLevel[EGraphicsPerfOverlayLevel["Minimal"] = 4] = "Minimal";
})(EGraphicsPerfOverlayLevel || (EGraphicsPerfOverlayLevel = {}));
export var EHDRToneMapOperator;
(function (EHDRToneMapOperator) {
    EHDRToneMapOperator[EHDRToneMapOperator["Invalid"] = 0] = "Invalid";
    EHDRToneMapOperator[EHDRToneMapOperator["Uncharted"] = 1] = "Uncharted";
    EHDRToneMapOperator[EHDRToneMapOperator["Reinhard"] = 2] = "Reinhard";
})(EHDRToneMapOperator || (EHDRToneMapOperator = {}));
export var ECPUGovernor;
(function (ECPUGovernor) {
    ECPUGovernor[ECPUGovernor["Invalid"] = 0] = "Invalid";
    ECPUGovernor[ECPUGovernor["Perf"] = 1] = "Perf";
    ECPUGovernor[ECPUGovernor["Powersave"] = 2] = "Powersave";
    ECPUGovernor[ECPUGovernor["Manual"] = 3] = "Manual";
})(ECPUGovernor || (ECPUGovernor = {}));
