// @decky/ui/dist/globals/steam-client/Notifications.js
export var EClientNotificationType;
(function (EClientNotificationType) {
    EClientNotificationType[EClientNotificationType["Invalid"] = 0] = "Invalid";
    EClientNotificationType[EClientNotificationType["DownloadComplete"] = 1] = "DownloadComplete";
    EClientNotificationType[EClientNotificationType["FriendInvite"] = 2] = "FriendInvite";
    EClientNotificationType[EClientNotificationType["FriendInGame"] = 3] = "FriendInGame";
    EClientNotificationType[EClientNotificationType["FriendOnline"] = 4] = "FriendOnline";
    EClientNotificationType[EClientNotificationType["Achievement"] = 5] = "Achievement";
    EClientNotificationType[EClientNotificationType["LowBattery"] = 6] = "LowBattery";
    EClientNotificationType[EClientNotificationType["SystemUpdate"] = 7] = "SystemUpdate";
    EClientNotificationType[EClientNotificationType["FriendMessage"] = 8] = "FriendMessage";
    EClientNotificationType[EClientNotificationType["GroupChatMessage"] = 9] = "GroupChatMessage";
    EClientNotificationType[EClientNotificationType["FriendInviteRollup"] = 10] = "FriendInviteRollup";
    EClientNotificationType[EClientNotificationType["FamilySharingDeviceAuthorizationChanged"] = 11] = "FamilySharingDeviceAuthorizationChanged";
    EClientNotificationType[EClientNotificationType["FamilySharingStopPlaying"] = 12] = "FamilySharingStopPlaying";
    EClientNotificationType[EClientNotificationType["FamilySharingLibraryAvailable"] = 13] = "FamilySharingLibraryAvailable";
    EClientNotificationType[EClientNotificationType["Screenshot"] = 14] = "Screenshot";
    EClientNotificationType[EClientNotificationType["CloudSyncFailure"] = 15] = "CloudSyncFailure";
    EClientNotificationType[EClientNotificationType["CloudSyncConflict"] = 16] = "CloudSyncConflict";
    EClientNotificationType[EClientNotificationType["IncomingVoiceChat"] = 17] = "IncomingVoiceChat";
    EClientNotificationType[EClientNotificationType["ClaimSteamDeckRewards"] = 18] = "ClaimSteamDeckRewards";
    EClientNotificationType[EClientNotificationType["GiftReceived"] = 19] = "GiftReceived";
    EClientNotificationType[EClientNotificationType["ItemAnnouncement"] = 20] = "ItemAnnouncement";
    EClientNotificationType[EClientNotificationType["HardwareSurvey"] = 21] = "HardwareSurvey";
    EClientNotificationType[EClientNotificationType["LowDiskSpace"] = 22] = "LowDiskSpace";
    EClientNotificationType[EClientNotificationType["BatteryTemperature"] = 23] = "BatteryTemperature";
    EClientNotificationType[EClientNotificationType["DockUnsupportedFirmware"] = 24] = "DockUnsupportedFirmware";
    EClientNotificationType[EClientNotificationType["PeerContentUpload"] = 25] = "PeerContentUpload";
    EClientNotificationType[EClientNotificationType["CannotReadControllerGuideButton"] = 26] = "CannotReadControllerGuideButton";
    EClientNotificationType[EClientNotificationType["Comment"] = 27] = "Comment";
    EClientNotificationType[EClientNotificationType["Wishlist"] = 28] = "Wishlist";
    EClientNotificationType[EClientNotificationType["TradeOffer"] = 29] = "TradeOffer";
    EClientNotificationType[EClientNotificationType["AsyncGame"] = 30] = "AsyncGame";
    EClientNotificationType[EClientNotificationType["General"] = 31] = "General";
    EClientNotificationType[EClientNotificationType["HelpRequest"] = 32] = "HelpRequest";
    EClientNotificationType[EClientNotificationType["OverlaySplashScreen"] = 33] = "OverlaySplashScreen";
    EClientNotificationType[EClientNotificationType["BroadcastAvailableToWatch"] = 34] = "BroadcastAvailableToWatch";
    EClientNotificationType[EClientNotificationType["TimedTrialRemaining"] = 35] = "TimedTrialRemaining";
    EClientNotificationType[EClientNotificationType["LoginRefresh"] = 36] = "LoginRefresh";
    EClientNotificationType[EClientNotificationType["MajorSale"] = 37] = "MajorSale";
    EClientNotificationType[EClientNotificationType["TimerExpired"] = 38] = "TimerExpired";
    EClientNotificationType[EClientNotificationType["ModeratorMsg"] = 39] = "ModeratorMsg";
    EClientNotificationType[EClientNotificationType["SteamInputActionSetChanged"] = 40] = "SteamInputActionSetChanged";
    EClientNotificationType[EClientNotificationType["RemoteClientConnection"] = 41] = "RemoteClientConnection";
    EClientNotificationType[EClientNotificationType["RemoteClientStartStream"] = 42] = "RemoteClientStartStream";
    EClientNotificationType[EClientNotificationType["StreamingClientConnection"] = 43] = "StreamingClientConnection";
    EClientNotificationType[EClientNotificationType["FamilyInvite"] = 44] = "FamilyInvite";
    EClientNotificationType[EClientNotificationType["PlaytimeWarning"] = 45] = "PlaytimeWarning";
    EClientNotificationType[EClientNotificationType["FamilyPurchaseRequest"] = 46] = "FamilyPurchaseRequest";
    EClientNotificationType[EClientNotificationType["FamilyPurchaseRequestResponse"] = 47] = "FamilyPurchaseRequestResponse";
    EClientNotificationType[EClientNotificationType["ParentalFeatureRequest"] = 48] = "ParentalFeatureRequest";
    EClientNotificationType[EClientNotificationType["ParentalPlaytimeRequest"] = 49] = "ParentalPlaytimeRequest";
    EClientNotificationType[EClientNotificationType["GameRecordingError"] = 50] = "GameRecordingError";
    EClientNotificationType[EClientNotificationType["ParentalFeatureResponse"] = 51] = "ParentalFeatureResponse";
    EClientNotificationType[EClientNotificationType["ParentalPlaytimeResponse"] = 52] = "ParentalPlaytimeResponse";
    EClientNotificationType[EClientNotificationType["RequestedGameAdded"] = 53] = "RequestedGameAdded";
    EClientNotificationType[EClientNotificationType["ClipDownloaded"] = 54] = "ClipDownloaded";
    EClientNotificationType[EClientNotificationType["GameRecordingStart"] = 55] = "GameRecordingStart";
    EClientNotificationType[EClientNotificationType["GameRecordingStop"] = 56] = "GameRecordingStop";
    EClientNotificationType[EClientNotificationType["GameRecordingUserMarkerAdded"] = 57] = "GameRecordingUserMarkerAdded";
    EClientNotificationType[EClientNotificationType["GameRecordingInstantClip"] = 58] = "GameRecordingInstantClip";
})(EClientNotificationType || (EClientNotificationType = {}));
export var ESystemUpdateNotificationType;
(function (ESystemUpdateNotificationType) {
    ESystemUpdateNotificationType[ESystemUpdateNotificationType["Invalid"] = 0] = "Invalid";
    ESystemUpdateNotificationType[ESystemUpdateNotificationType["Available"] = 1] = "Available";
    ESystemUpdateNotificationType[ESystemUpdateNotificationType["NeedsRestart"] = 2] = "NeedsRestart";
})(ESystemUpdateNotificationType || (ESystemUpdateNotificationType = {}));
export var EGameRecordingErrorType;
(function (EGameRecordingErrorType) {
    EGameRecordingErrorType[EGameRecordingErrorType["General"] = 1] = "General";
    EGameRecordingErrorType[EGameRecordingErrorType["LowDiskSpace"] = 2] = "LowDiskSpace";
})(EGameRecordingErrorType || (EGameRecordingErrorType = {}));
