// @decky/ui/dist/utils/static-classes.js
import { findClassModule } from '../class-mapper';
export const quickAccessMenuClasses = findClassModule((m) => m.Title && m.QuickAccessMenu && m.BatteryDetailsLabels);
export const scrollPanelClasses = findClassModule((m) => m.ScrollPanel);
export const gamepadDialogClasses = findClassModule((m) => m.GamepadDialogContent && !m.BindingButtons);
export const quickAccessControlsClasses = findClassModule((m) => m.BatteryPercentageLabel && m.PanelSection && !m['vr-dashboard-bar-height'] && !m.QuickAccessMenu && !m.QuickAccess && !m.PerfProfileInfo);
export const updaterFieldClasses = findClassModule((m) => m.OOBEUpdateStatusContainer);
export const playSectionClasses = findClassModule((m) => m.PlayBarDetailLabel);
export const gamepadSliderClasses = findClassModule((m) => m.SliderControlPanelGroup);
export const appDetailsHeaderClasses = findClassModule((m) => m.TopCapsule);
export const appDetailsClasses = findClassModule((m) => m.HeaderLoaded);
export const gamepadUIClasses = findClassModule((m) => m.BasicUiRoot);
export const gamepadTabbedPageClasses = findClassModule((m) => m.GamepadTabbedPage);
export const gamepadContextMenuClasses = findClassModule((m) => m.BasicContextMenuModal);
export const achievementListClasses = findClassModule((m) => m.AchievementListItemBase && !m.Page);
export const achievementPageClasses = findClassModule((m) => m.AchievementListItemBase && m.Page);
export const mainMenuAppRunningClasses = findClassModule((m) => m.AppRunningControls && m.OverlayAchievements);
export const basicAppDetailsSectionStylerClasses = findClassModule((m) => m.AppDetailsRoot);
export const steamSpinnerClasses = findClassModule(m => m.SpinnerLoaderContainer);
export const footerClasses = findClassModule(m => m.QuickAccessFooter);
export const appActionButtonClasses = findClassModule(m => m.PlayButtonContainer);
export const libraryAssetImageClasses = findClassModule(m => m.LongTitles && m.GreyBackground);
export const gamepadLibraryClasses = findClassModule(m => m.GamepadLibrary);
export const focusRingClasses = findClassModule(m => m.FocusRingRoot);
export const searchBarClasses = findClassModule(m => m.SearchAndTitleContainer);
export const mainBrowserClasses = findClassModule(m => m.MainBrowserContainer);
export const staticClasses = quickAccessMenuClasses;
export const scrollClasses = scrollPanelClasses;
export const achievementClasses = achievementListClasses;
